src/Compilers/CSharp/Portable/Symbols/Metadata/PE/PEAssemblySymbol.cs

```diff
@@ -280,5 +283,17 @@ internal PEModuleSymbol PrimaryModule
         }
 
         public override AssemblyMetadata GetMetadata() => _assembly.GetNonDisposableMetadata();
+
+        internal bool GetCompilerFeatureRequiredUseSiteInfo(ref UseSiteInfo<AssemblySymbol> result, MetadataDecoder decoder)
+        {
+            if (_lazyCachedUseSiteInfo.IsInitialized)
+            {
+                return MergeUseSiteInfo(ref result, _lazyCachedUseSiteInfo.ToUseSiteInfo(PrimaryDependency));
+            }
+
+            PEUtilities.DeriveUseSiteInfoFromCompilerFeatureRequiredAttributes(ref result, this, PrimaryModule, this.Assembly.Handle, CompilerFeatureRequiredFeatures.None, decoder);
+            _lazyCachedUseSiteInfo.Initialize(result.PrimaryDependency, result);
```

Comment:
>_lazyCachedUseSiteInfo.Initialize(result.PrimaryDependency, result); [](http://example.com/codeflow?start=12&length=68)

I am having a very hard time trying to understand what  we are trying to accomplish here. Why would we want to cache something based on the input parameter? Is there a guarantee that the input is always the same? Either I am missing something or things are made more complicated than they could be. At the moment, I think that there is no reason for us to cache a full use site info for an assembly symbol (and for a module symbol as well), we never really need to ask for it. So, why deal with a more complicated object, ```CachedUseSiteInfo<AssemblySymbol>```, deal with dependencies, etc.? I think that it would be sufficient to have an API that returns cached string with required feature name, if any. Or, if we really want to reuse a DiagnosticInfo object, an API that returns that cached DiagnosticInfo.